config/trace.go

```diff
@@ -4,15 +4,172 @@
 package config // import "go.opentelemetry.io/contrib/config"
 
 import (
+	"context"
+	"errors"
+	"fmt"
+	"net/url"
+	"time"
+
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracehttp"
+	"go.opentelemetry.io/otel/exporters/stdout/stdouttrace"
+	"go.opentelemetry.io/otel/sdk/resource"
 	sdktrace "go.opentelemetry.io/otel/sdk/trace"
 	"go.opentelemetry.io/otel/trace"
 	"go.opentelemetry.io/otel/trace/noop"
 )
 
-func initTracerProvider(cfg configOptions) (trace.TracerProvider, shutdownFunc) {
+func tracerProvider(cfg configOptions, res *resource.Resource) (trace.TracerProvider, shutdownFunc, error) {
 	if cfg.opentelemetryConfig.TracerProvider == nil {
-		return noop.NewTracerProvider(), noopShutdown
+		return noop.NewTracerProvider(), noopShutdown, nil
+	}
+	opts := []sdktrace.TracerProviderOption{
+		sdktrace.WithResource(res),
+	}
+	for _, processor := range cfg.opentelemetryConfig.TracerProvider.Processors {
+		sp, err := spanProcessor(cfg.ctx, processor)
+		if err != nil {
+			return noop.NewTracerProvider(), noopShutdown, err
+		}
```

Comment:
Suggestion: Nothing is worse than fixing one error for a new one to appear.  Could we move the return the error case to be something that combines all the errors?

Reply:
@MadVikingGod i've updated this to create a slice of errs and return using errors.Join if it's not empty, please take a look